src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package mate.academy.service.impl;
+
+import java.util.List;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.dao.TicketDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private TicketDao ticketDao;
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        ShoppingCart shoppingCart = getByUser(user);
+        shoppingCart.getTicketList().add(ticketDao.add(new Ticket(movieSession, user)));
+        shoppingCartDao.update(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        return shoppingCartDao.getByUser(user).orElseThrow(()
+                -> new RuntimeException("Can't find shopping cart by user: " + user));
+    }
+
+    @Override
+    public void registerNewShoppingCart(User user) {
+        shoppingCartDao.add(new ShoppingCart(List.of(), user));
```

Comment:
```suggestion
        ShoppingCart shoppingCart = new ShoppingCart(new ArrayList(), user)
        shoppingCartDao.add(shoppingCart);
```
to increase readability
also why list.of() when you can just pass new list